### PR TITLE
Gradient Animations

### LINKED ISSUE #119: Gradient Animations (preview)
XAML api for gradient animation. Planned features:

- animate any property on any Gradient Element
- use from Behaviour or Trigger Action
- single animation and Storyboard (parent animation with child animations, child can set position withing parent storyboard)
- different repeat modes (repeat x-times or forever), auto reverse (animate backwards)
- use simple From-To and Key Frames

### PR DESCRIPTION
**Pull request by mgierlasinski**
Resolves #119

### CODE REVIEW

File: PlaygroundLite/PlaygroundLite/Controls/MenuSection.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+
+namespace PlaygroundLite.Controls
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
```
Comment (bbenetskyy) [inline PlaygroundLite/PlaygroundLite/Controls/MenuSection.xaml.cs:12]: we have it at `AssemblyInfo.cs` and could safely remove it

File: PlaygroundLite/PlaygroundLite/Pages/MainPage.xaml [hunk 2]
```diff
@@ -1,14 +1,26 @@
 ﻿<?xml version="1.0" encoding="utf-8" ?>
 <ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
              xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:controls="clr-namespace:PlaygroundLite.Controls;assembly=PlaygroundLite"
              x:Class="PlaygroundLite.Pages.MainPage"
              Title="Test your might">
 
     <ScrollView>
         <StackLayout Margin="10">
+            <controls:MenuSection>
```
Comment (bbenetskyy) [inline PlaygroundLite/PlaygroundLite/Pages/MainPage.xaml:10]: I love how it looks now 🤘🤘🤘

**Review by bbenetskyy — APPROVED** [review]

File: MagicGradients/Animation/Interactivity/Animate.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            _associatedObject = bindable;
+
+            if (Animation == null)
+                return;
+
+            if (Animation.Target == null)
+                Animation.Target = _associatedObject;
+
+            // TODO: is it required?
+            await Task.Delay(250);
```
[the remark this answers is not in the document]
Comment (mgierlasinski) [reply MagicGradients/Animation/Interactivity/Animate.cs:24]: I still don't know :) Forgot to check, this was borrowed from other Animation library.

[the remark this answers is not in the document]
Comment (bbenetskyy) [reply MagicGradients/Animation/Interactivity/Animate.cs:24]: Then I assume that this is required old evil and if we remove it something definitely will break later

[the remark this answers is not in the document]
Comment (mgierlasinski) [reply MagicGradients/Animation/Interactivity/Animate.cs]: I did it better way :) Closing.

**Pull request merged**